bundles/org.openhab.automation.jythonscripting/src/main/java/org/openhab/automation/jythonscripting/JythonScriptEngineFactory.java

```diff
@@ -0,0 +1,132 @@
+                .toString().replace("file:", "");
+        System.setProperty(PYTHON_HOME, pythonHome);
+
+        String existingPythonPath = System.getProperty(PYTHON_PATH);
+        if (existingPythonPath == null || existingPythonPath.isEmpty()) {
+            System.setProperty(PYTHON_PATH, DEFAULT_PYTHON_PATH);
+        } else if (!existingPythonPath.contains(DEFAULT_PYTHON_PATH)) {
+            Set<String> newPythonPathList = new TreeSet<>(
+                    new ArrayList<>(Arrays.asList(existingPythonPath.split(File.pathSeparator))));
+            newPythonPathList.add(DEFAULT_PYTHON_PATH);
+            String newPythonPath = String.join(File.pathSeparator, newPythonPathList);
+            System.setProperty(PYTHON_PATH, newPythonPath);
+        }
+
+        System.setProperty(PYTHON_CACHEDIR,
+                Paths.get(OpenHAB.getUserDataFolder(), "cache", "org.openhab.automation.jythonscripting", "cachedir")
+                        .toString());
+
+        logPythonPaths();
+    }
+
+    private void logPythonPaths() {
+        logger.trace("{}: {}, {}: {}, {}: {}", //
+                PYTHON_HOME, System.getProperty(PYTHON_HOME), //
+                PYTHON_PATH, System.getProperty(PYTHON_PATH), //
+                PYTHON_CACHEDIR, System.getProperty(PYTHON_CACHEDIR));
+    }
+
+    @Override
+    public List<String> getScriptTypes() {
+        List<String> scriptTypes = new ArrayList<>();
+
+        for (javax.script.ScriptEngineFactory factory : ENGINE_MANAGER.getEngineFactories()) {
+            List<String> extensions = factory.getExtensions();
+
+            if (extensions.contains(SCRIPT_TYPE)) {
+                scriptTypes.addAll(extensions);
+                scriptTypes.addAll(factory.getMimeTypes());
+            }
+        }
+        return List.copyOf(scriptTypes);
+    }
+
+    @Override
+    public @Nullable ScriptEngine createScriptEngine(String scriptType) {
+        ScriptEngine scriptEngine = ENGINE_MANAGER.getEngineByExtension(scriptType);
+        if (scriptEngine == null) {
+            scriptEngine = ENGINE_MANAGER.getEngineByMimeType(scriptType);
+        }
+        if (scriptEngine == null) {
+            scriptEngine = ENGINE_MANAGER.getEngineByName(scriptType);
+        }
+        return scriptEngine;
+    }
+
+    @Deactivate
+    public void removePythonPath() {
+        logger.debug("Unloading JythonScriptEngineFactory");
+
+        String existingPythonPath = System.getProperty(PYTHON_PATH);
+        if (existingPythonPath != null && existingPythonPath.contains(DEFAULT_PYTHON_PATH)) {
+            Set<String> newPythonPathList = new TreeSet<>(
+                    new ArrayList<>(Arrays.asList(existingPythonPath.split(File.pathSeparator))));
+            newPythonPathList.remove(DEFAULT_PYTHON_PATH);
+            String newPythonPath = String.join(File.pathSeparator, newPythonPathList);
+            System.setProperty(PYTHON_PATH, newPythonPath);
```

Comment:
This looks to be duplicate code from above.

Reply:
Not exactly. The difference is that one adds and the other removes the path.